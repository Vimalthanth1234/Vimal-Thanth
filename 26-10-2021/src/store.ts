import { createStore,compose } from "redux";
import rootReducer from "./redux/rootReducer";
declare global {
    interface Window {
      __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
    }
  }
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(rootReducer, composeEnhancers());
// const store = createStore(rootReducer , window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__())

export default store