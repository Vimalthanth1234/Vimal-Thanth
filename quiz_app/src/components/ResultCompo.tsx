import React from 'react'
import { useSelector } from 'react-redux'
import {useState} from 'react'
import { PieChart } from 'react-minimal-pie-chart';
type State = {
    startReducer:string,
    allQuestionsReducer:any,
    quesNumReducer:number
}
const ResultCompo = () => {
    const myState = useSelector((state:State)=>{
        return state
    })
    const questions = myState.allQuestionsReducer[myState.startReducer]
    const result:any = []
    questions.map((ele:any)=>{
        if(ele.type!=='Multi Select'){
            if(ele.Answer[0]==ele.selectedAns[0]){
                console.log(ele.selectedAns[0])
                result.push(true)
            }
        }else{
            const multiArr:any = []
            ele.selectedAns.map((e:string)=>{
                console.log(e)
                if(ele.Answer.includes(e)){
                    multiArr.push(true)
                }else{
                    multiArr.push(false)
                }
            })
            if(multiArr.length){
                if(multiArr.every((val:boolean)=>val===true)){
                    if(multiArr.length===ele.Answer.length){
                        console.log(ele.Answer.length)
                        console.log(multiArr.length)
                        result.push(true)
                        console.log(result)
                    }
                }
            }
        }
    })
    return (
        <div style={{boxShadow:'5px 10px #888888',padding:'50px'}}>
            <h1 style={{color:'blue',textAlign:'center',boxShadow:'5px 10px #888888',padding:'10px'}}>Result</h1>
        <div style={{display:'flex',flexDirection:'row'}}>
            <div>
            <PieChart style={{display:'block'}}
                data={[
                { title: 'Right', value: result.length, color: 'green' },
                { title: 'Wrong', value: questions.length-result.length, color: 'red' }
            ]}
            />
            </div>
            <div style={{display:'flex',justifyContent:'center',alignItems:'center',flexDirection:'column',margin:'10px'}}>
            <p style={{color:'green',textAlign:'center'}}>True Answers :- {result.length}</p>
            <p style={{color:'red',textAlign:'center'}}>False Answers :- {questions.length-result.length}</p>
            <p style={{textAlign:'center'}}>Percentage :- {(result.length/questions.length)*100}%</p>
            </div>
        </div>
        </div>
    )
}
export default ResultCompo
