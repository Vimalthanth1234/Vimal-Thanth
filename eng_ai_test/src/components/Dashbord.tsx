import React, { useState } from 'react'
import Navbar from './Navbar'
import { useSelector,useDispatch } from 'react-redux'  
import isAnswered from '../redux/actions/isAnswered' 
import selectedAnswer from '../redux/actions/selectedAnswer'
import ResultCompo from './ResultCompo'
import next from '../redux/actions/next'
import previous from '../redux/actions/previous'
import Button from '@mui/material/Button'
import quesNumReducer from '../redux/reducers/quesNumReducer'
import { Link } from 'react-router-dom'
type State = {
    startReducer:string,    
    allQuestionsReducer:any
    quesNumReducer:number
}
const Dashbord = () => {
    const myState = useSelector((state:State)=>{
        return state
    })
    const dispatch = useDispatch()
    const questions = myState.allQuestionsReducer[myState.startReducer]
    const question:any = questions[myState.quesNumReducer]
    const [check,setCheck]:any = useState(question.selectedAns)
    const compo = ()=>{
        switch (question.type) {
            case 'Fill In The Blanks':return(
                <>
                <h4> {question.question}</h4>
                <ul>
                    {question.options.map((ele:string,index:number)=>{
                        return <><li><input type='radio' onClick={()=>{
                            question.type!=='Multi Select'&&dispatch(selectedAnswer(myState.startReducer,myState.quesNumReducer,[ele]))
                            setCheck(question.selectedAns[0])
                            dispatch(isAnswered(myState.startReducer,myState.quesNumReducer,true))
                        }} name='options1' checked={question.selectedAns==ele?true:false}/>{ele}</li></>
                    })}
                </ul>
                </>
            )
            case 'MCQ':return(
                <>
                <h4>{question.question}</h4>
                <ul>
                    {question.options.map((ele:string)=>{
                        return <><li><input type='radio' name='options2' onClick={()=>{
                            question.type!=='Multi Select'&&dispatch(selectedAnswer(myState.startReducer,myState.quesNumReducer,[ele]))
                            setCheck(question.selectedAns[0])
                            dispatch(isAnswered(myState.startReducer,myState.quesNumReducer,true))
                        }} checked={question.selectedAns==ele?true:false} />{ele}</li></>
                    })}
                </ul>
                </>
            )
            case 'True Or False':return(
                <>
                <h4>{question.question}</h4>
                <ul>
                    {question.options.map((ele:string)=>{
                        return <><li><input type='radio' onClick={()=>{
                            question.type!=='Multi Select'&&dispatch(selectedAnswer(myState.startReducer,myState.quesNumReducer,[ele]))
                            setCheck(question.selectedAns[0])
                            dispatch(isAnswered(myState.startReducer,myState.quesNumReducer,true))
                        }} name='options3' checked={question.selectedAns==ele?true:false} />{ele}</li></>
                    })}
                </ul>
                </>
            )
            case 'Match The Following':return(
                <>
                <table>
                    <thead>
                        <tr>
                            <th>Questions</th>
                            <th>Answers</th>
                        </tr>
                    </thead>
                    <tbody>
                        {question.question[0].map((ele:any,index:number)=>{
                            return <><tr><td>{question.question[0][index]}</td><td>{question.question[1][index]}</td></tr></>
                        })}
                    </tbody>
                </table>
                <ul>
                    {question.options.map((ele:string)=>{
                        return <><li><input type='radio' onClick={()=>{
                            dispatch(isAnswered(myState.startReducer,myState.quesNumReducer,true))
                            setCheck(question.selectedAns[0])
                            question.type!=='Multi Select'&&dispatch(selectedAnswer(myState.startReducer,myState.quesNumReducer,[ele]))
                        }} name='options4' checked={question.selectedAns==ele?true:false} />{ele}</li></>
                    })}
                </ul>
                </>
            )
            case 'Multi Select':return(
                <>
                <h4>{question.question}</h4>
                <ul>
                    {question.options.map((ele:string,index:number)=>{
                        return <><li><input value={ele} type='checkbox' onClick={(event:any)=>{
                            let selectedValues = question.selectedAns
                            if(selectedValues.includes(event.target.value)){
                                const index = selectedValues.indexOf(event.target.value)
                                selectedValues.splice(index,1)
                                check?setCheck(''):setCheck('hello')
                                dispatch(selectedAnswer(myState.startReducer,myState.quesNumReducer,selectedValues))
                            }else{
                                selectedValues.push(event.target.value)
                                setCheck(ele)
                                dispatch(selectedAnswer(myState.startReducer,myState.quesNumReducer,selectedValues))
                            }
                            question.selectedAns.length?dispatch(isAnswered(myState.startReducer,myState.quesNumReducer,true)):dispatch(isAnswered(myState.startReducer,myState.quesNumReducer,false))
                        }} name='options5' checked={question.selectedAns.includes(ele)?true:false}/>{ele}</li></>
                    })}
                </ul>
                </>
            ) 
            default:
                break;
        }
    }
    return (
        <div style={{boxShadow:'5px 10px #888888',padding:'50px'}}>
            <h1 style={{textAlign:'center',boxShadow:'5px 10px #888888',padding:'10px'}}><span style={{color:'purple'}}>Q</span>UI<span style={{color:'blue'}}>Z</span></h1>
            <Navbar />
            <h3>Ques No :- {question['Question Number']}</h3>
            {compo()}
            <Button style={{backgroundColor:'blue',color:'white'}} onClick={()=>dispatch(previous())} disabled={myState.quesNumReducer==0?true:false}>Previous</Button>
            <Button style={{backgroundColor:'purple',color:'white'}} onClick={()=>dispatch(next())} disabled={questions.length-1==myState.quesNumReducer?true:false}>Next</Button>
            {myState.quesNumReducer==questions.length-1&&<Button variant='contained'><Link to='/resultcompo' style={{textDecoration:'none',color:'white'}}>Submit test</Link></Button>}
        </div>
    )
}
export default Dashbord
