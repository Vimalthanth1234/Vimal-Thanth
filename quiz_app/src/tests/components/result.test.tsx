import Dashbord from '../../components/Dashbord'
import * as ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import ResultCompo from '../../components/ResultCompo'
import {Provider} from 'react-redux'
import store from '../../redux/store'
import React from 'react'
describe('Home Component tests',()=>{
    let container:HTMLDivElement
    beforeEach(()=>{
        container = document.createElement('div');
        document.body.appendChild(container);
        ReactDOM.render(
            <React.StrictMode>
            <BrowserRouter>
            <Provider store={store} >
            <ResultCompo />
            </Provider></BrowserRouter></React.StrictMode>,container)
    })
    afterEach(()=>{
        document.body.removeChild(container);
        container.remove()
    })
    it('Render correctly initial document',()=>{
        expect(container).toMatchSnapshot()
    })
})