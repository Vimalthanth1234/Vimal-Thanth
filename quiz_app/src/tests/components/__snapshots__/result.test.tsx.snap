// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component tests Render correctly initial document 1`] = `
<div>
  <div
    style="box-shadow: 5px 10px #888888; padding: 50px;"
  >
    <h1
      style="color: blue; text-align: center; box-shadow: 5px 10px #888888; padding: 10px;"
    >
      Result
    </h1>
    <div
      style="display: flex; flex-direction: row;"
    >
      <div>
        <svg
          height="100%"
          style="display: block;"
          viewBox="0 0 100 100"
          width="100%"
        >
          <path
            d=""
            fill="none"
            stroke="green"
            stroke-width="50"
          >
            <title>
              Right
            </title>
          </path>
          <path
            d="M 75 50 A 25 25 0 1 1 74.99999999619229 49.99956366768703"
            fill="none"
            stroke="red"
            stroke-width="50"
          >
            <title>
              Wrong
            </title>
          </path>
        </svg>
      </div>
      <div
        style="display: flex; justify-content: center; align-items: center; flex-direction: column; margin: 10px;"
      >
        <p
          style="color: green; text-align: center;"
        >
          True Answers :- 
          0
        </p>
        <p
          style="color: red; text-align: center;"
        >
          False Answers :- 
          5
        </p>
        <p
          style="text-align: center;"
        >
          Percentage :- 
          0
          %
        </p>
      </div>
    </div>
  </div>
</div>
`;
